app/reducers/surveySubmissions.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import { SurveySubmission } from '../actions/ActionTypes';
+import createEntityReducer from 'app/utils/createEntityReducer';
+import { createSelector } from 'reselect';
+import type { OptionEntity, SurveyEntity, QuestionEntity } from './surveys';
+import type { UserEntity } from 'app/reducers/users';
+
+export type SubmissionEntity = {
+  id: number,
+  user: UserEntity,
+  survey: SurveyEntity,
+  submitted: boolean,
+  submittedTime?: string,
+  answers: Array<AnswerEntity>
+};
+
+export type AnswerEntity = {
+  id: number,
+  submission: SubmissionEntity,
+  question: QuestionEntity,
+  answerText: string,
+  selectedOptions: Array<OptionEntity>
+};
+
+function mutateSurveySubmissions(state, action) {
+  switch (action.type) {
+    case SurveySubmission.FETCH.SUCCESS: {
+      return {
+        ...state,
+        items: state.items.filter(id => id !== action.meta.id)
+      };
+    }
+
+    default:
+      return state;
+  }
+}
+
+export default createEntityReducer({
+  key: 'surveySubmissions',
+  types: {
+    fetch: SurveySubmission.FETCH,
+    mutate: SurveySubmission.ADD
+  },
+  mutate: mutateSurveySubmissions
+});
+
+export const selectSurveySubmissions = createSelector(
+  (state, props) => props.surveyId,
+  state => state.surveySubmissions.items,
+  state => state.surveySubmissions.byId,
+  state => state.users.byId,
+  (surveyId, surveySubmissionIds, surveySubmissionsById, usersById) =>
+    surveySubmissionIds
+      .map(surveySubmissionId => {
```

Comment:
Hmmm. This will map over all submissions in the state, and then filter... With the current implementation, this looks like the only solution, but i guess we can do better?

Maybe we can:
- Add a list of all `submissionIds` into the submission.
- Or maybe do something like this :thinking:  https://github.com/webkom/lego-webapp/blob/106e835d7f08d4f845e0e74bf985ba93832bae42/app/reducers/index.js#L176